tests/test_server.py

```diff
@@ -220,6 +221,9 @@ def test_can_run_main(self, ioloop_mock, get_server_parameters_mock, server_mock
         main()
         ioloop_instance_mock.start.assert_any_call()
 
+    def cleanup(self):
+        ServerTestCase.cleanup_called = True
+
     @mock.patch.object(thumbor.server, 'HTTPServer')
     @mock.patch.object(thumbor.server, 'get_server_parameters')
     @mock.patch('tornado.ioloop.IOLoop.instance', create=True)
```

Comment:
Tried it, doesn't work. You're welcome to give it a try, I wasted 2-3 hours on this yesterday.

It might look like it works at some point, but try replacing assert_called with assert_not_called and both pass, which is nonsensical.

I think this is because the engine (and as a side-effect, BaseEngine) is imported programmatically by importer.py. Mock is made for imports that are declared. I had the impression that the workaround was to mock.patch  __import__ itself to catch the dynamic import, etc. but then imho that's even uglier and complicated than my solution.